algviz/linkedlist/linkedlistvisualizer.py

```diff
@@ -1,152 +1,152 @@
-	def __init__(self, svg_doc = None):
-		self.svg_doc = svg_doc
-
-	def _draw(self, ll, center = None, nested = False, getSize = False):
-		self.draw_contents(item = ll)
-
-if __name__ == "__main__":
-	ll = LinkedList()
-	ll2 = LinkedList()
-	n7 = Node(4, None)
-	n6 = Node(111, n7)
-	n8 = Node(0, n6)
-	ll2.addNode(n8)
-	n5 = Node(321, None)
-	n4 = Node(144123, n5)
-	n3 = Node(ll2, n4)
-	n2 = Node(2232, n3)
-	n1 = Node(123, n2)
-	ll.addNode(n1)
-
-
-	t = LLVisualiser()
-	t._draw(ll)
+import svgwrite
+class Node:
+	def __init__(self, value = None, nextnode = None):
+		self.value = value
+		self.nextnode = nextnode
+
+	def setValue(self, value):
+		self.value = value
+
+	def setNext(self, nextnode):
+		self.nextnode = nextnode
+
+class LinkedList:
+	def __init__(self, start = None):
+		self.start = start
+
+	def addNode(self, node):
+		current = self.start
+		if(self.start is None):
+			self.start = node
+		else:
+			while(current.nextnode is not None):
+				current = current.nextnode
+			current.nextnode = node#add the node
+
+#	def length(self):
+#		current = self.start
+#		count = 0;
+#		while(current is not None):
+#			count += 1
+#			current = current.nextnode
+#		return count
+
+	def __iter__(self):
+		self.i = self.start
+		return self
+
+	def __next__(self):
+		if self.i is None:
+			raise StopIteration
+		else:
+			item = self.i.value
+			self.i = self.i.nextnode
+			return item
+
+class BaseVisualizer:
+	def __init__(self, svg_doc = None):
+		self.svg_doc = svg_doc
+
+	def save(self):
+		if self.svg_doc is not None:
+			self.svg_doc.save()
+
+	def length(self, item):
+		length = 0
+		for i in item:
+			length += 1
+		return length
+
+	def draw_contents(self, item = None, svg_doc = None, center = None, nested = False, getSize = False):#universal drawer; figures out which function to use to draw the requested item
+		if type(item) == int:
+			return self.draw_string(item, svg_doc, center, nested, getSize)
+		elif type(item) == LinkedList:
+			return self.draw_linkedlist(item, svg_doc, center, nested, getSize)
+
+	def draw_string(self, item, svg_doc = None, center = None, nested = False, getSize = False):
+		size = 30 * len(str(item))#might be too complicated to have this be passed as an arg between funcs
+		offset = 0
+		if nested:
+			offset = size / 2
+		if not getSize:#draw unless we're getting the size	
+			svg_doc.add(svg_doc.text(text = str(item), insert = [center[0] - offset/2, center[1] + (offset / len(str(item)) / 2)], font_size = 30))
+		return size
+
+	def draw_linkedlist(self, ll, svg_doc = None, center = None, nested = False, getSize = False):
+		size = 0
+		length = self.length(ll)
+		offset_array = []
+		radius = 40#default radius size
+		if(self.svg_doc is None and not getSize):#note, default length can be overflowed as it stands, need to come up with a better way of determining length
+			center = [radius * 2.5, 200]
+			value = self.draw_contents(ll, self.svg_doc, center, getSize = True)#how long should our document be?
```

Comment:
I know it's not your code and you don't want to rewrite it now during finals week.  I like that you're moving it out of the root directory of our project.  But this code should live in a directory full of very rough prototypes, not in the core of our module.